examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+   "metadata": {},
+   "source": [
+    "This works like a charm!  \n",
+    "Next, we **convert this into a widget.**  \n",
+    "For that, we need the code in a _ems string inside a class that inherits from `ipyreact.ReactWidget`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c70385c-638e-482b-80e1-6f27614da867",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton() {\n",
+    "        return < button > X < /button> \n",
+    "    };\"\"\"\n",
+    "\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "35f22aef-50c2-4772-b9fa-44f5fb486ee6",
+   "metadata": {},
+   "source": [
+    "now it's time to make some **styling using CSS**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38e8afeb-274c-4c62-99a7-6200fc9c179d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "css_rules = \"\"\"\n",
+    "button {\n",
+    "    background-color: yellow;\n",
+    "    border-radius: 10px;\n",
+    "}\n",
+    "\"\"\"\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb19480f-b3c6-44be-968c-06edf944d42f",
+   "metadata": {},
+   "source": [
+    "You can also load CSS from a file.  \n",
+    "And you can add extra CSS without overwriting the previous CSS import."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c4a644e0-1dbe-4f50-9bb5-5a54f1fedbaf",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "from pathlib import Path\n",
+    "css_rules = Path(\"styles_orange.css\").read_text()\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f571eadd-1a94-4ffe-af43-1c5ef1ec2498",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Note: When you clear the output of the previous cell, this CSS style will be removed from the notebook."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "b50fbd27-5679-48ea-b2cc-ff9c5c79f826",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Next, we want to **add parameters to the widget.**  \n",
+    "First naive approach: Using f-strings.\n",
+    "That works, but is bad for two reasons:\n",
+    "1. curly brackets from f-strings will collide with the curly brackets from css. \n",
+    "Using curly brackets in an f-string as a normal character can be done by doubling them like this : `{`to `{{`.\n",
+    "2. It is not possible to pass this parameter into the `MyExampleWidget` class, as it does not have an init function.\n",
```

Comment:
This is not correct, it does have an `__init__` (well, it's super class), but that does not matter. The f-string is only interpreted when the class sourcecode is parsed/interpreted, not at "runtime", e.g. when you create an object, nor would it be possible to have different objects with different `my_message` values.
Hope this helps you in rephrasing what you want to say.